trl/commands/cli_utils.py

```diff
@@ -172,6 +172,59 @@ class DpoScriptArguments:
     )
 
 
+@dataclass
+class ChatArguments:
+    # general settings
+    model: str = field(metadata={"help": "Name of the pre-trained model"})
+    user: str = field(default=None, metadata={"help": "Username to display in chat interface"})
+    system_prompt: str = field(default=None, metadata={"help": "System prompt"})
+    save_folder: str = field(default="./chat_history/", metadata={"help": "Folder to save chat history"})
+    device: str = field(default="cpu", metadata={"help":"device to use for inference."},)
+    # generation settings
+    max_new_tokens: int = field(default=256, metadata={"help": "Maximum number of tokens to generate"})
+    do_sample: bool = field(default=True, metadata={"help": "Whether to sample outputs during generation"})
+    num_beams: int = field(default=1, metadata={"help": "Number of beams for beam search"})
+    temperature: float = field(default=1.0, metadata={"help": "Temperature parameter for generation"})
+    top_k: int = field(default=50, metadata={"help": "Value of k for top-k sampling"})
+    top_p: float = field(default=1.0, metadata={"help": "Value of p for nucleus sampling"})
+    repetition_penalty: float = field(default=1.0, metadata={"help": "Repetition penalty"})
+    # model loading
+    model_revision: str = field(
```

Comment:
I think for all the arguments below you could leverage `ModelConfig` from trl instead and on the script above declare a `TrlParser` with `ChatArguments` and `ModelConfig` - see: https://github.com/huggingface/trl/blob/main/examples/scripts/sft.py#L85-L101

Reply:
I was wondering the same, but it contains also a lot of args that are only relevant to training which i think is confusing (and would show up with help, no?).

Reply:
ah yes good point ! then i think all good